drivers/gpio/gpio_intel_apl.c

```diff
@@ -30,9 +31,12 @@
 
 #include "gpio_utils.h"
 
-#ifndef CONFIG_SHARED_IRQ
-#error "Need CONFIG_SHARED_IRQ!"
-#endif
+/*
+ * only IRQ 14 is supported now. the docs say IRQ 15 is supported
```

Comment:
IIRC, a few registers need to be manipulated for IRQ 15 to work. So I would prefer letting the developers to decide instead of asserting here.

Reply:
Are you saying I should remove the build assertion, or remove my assertion in the comments that IRQ 15 doesn't work?  I think the build assertion should stay, because the driver really does only work if IRQ 14 is selected, so devicetree should reflect that.  I can reword the comment if you think I'm being unfair to the documentation :)

Reply:
I forgot the part where IRQ 14 is forced. So everything's fine.